src/comments.js

```diff
@@ -381,6 +383,20 @@ function handleConditionalExpressionComments(enclosingNode, followingNode, comme
   return false;
 }
 
+function handleAssignmentPatternComments(enclosingNode, comment) {
```

Comment:
I did investigate a bit. For the following case, this line is involved https://github.com/jlongster/prettier/blob/master/src/comments.js#L192. Then `addCommentHelper` enforces`comment.printed = false` which strips the comment. If set to *true* then this is working again:

```js
const { c = 1 /* comment */ } = d;
```
I don't think you want to alter such a primitive function, that's why I used this `handleAssignmentPatternComments` function.